### PR TITLE
Replaces Black with Ruff Format

### PR DESCRIPTION
**Pull request by kjaymiller**
This pull request removes the Black package from the project dependencies and replaces it with Ruff for code formatting. 

The code has been updated to use the `ruff lint` and `ruff format` commands instead of running Black. This change fixes issue #259.

### CODE REVIEW

File: {{cookiecutter.__src_folder_name}}/.github/workflows/format.yml [hunk 1]
```diff
@@ -25,6 +25,6 @@ jobs:
             python -m pip install --upgrade pip
             pip install -r requirements-dev.in
         - name: Lint with ruff
-          run: ruff .
-        - name: Check formatting with black
-          run: black . --check --verbose
+          run: |
+            ruff lint .
```
Comment (kjaymiller) [inline {{cookiecutter.__src_folder_name}}/.github/workflows/format.yml:29]:
```suggestion
            ruff check .
```

File: tools/update_info.py [hunk 2]
```diff
@@ -99,6 +99,8 @@ def update_all_repos():
         )
         subprocess.check_output(
             ["git", "push", "--set-upstream", "origin", random_cc_folder_prefix],
+            text=True,
```
Comment (pamelafox) [inline tools/update_info.py:102]: Curious why these arguments are now needed?

File: {{cookiecutter.__src_folder_name}}/requirements-dev.in [hunk 3]
```diff
@@ -17,5 +17,4 @@ pytest-flask
 pytest-playwright
 
 # Linters
-ruff
-black
+ruff
```
Comment (pamelafox) [inline {{cookiecutter.__src_folder_name}}/requirements-dev.in:20]: EOF newline changed.

We do have a precommit that fixes EOF newlines in the root of this project, I believe.

File: hooks/post_gen_project.py [hunk 4]
```diff
@@ -120,30 +120,26 @@ def check_for_files() -> None:
...
     """Runs all linters"""
-    run_ruff_fix_and_black()
+    run_ruff_lint_and_format()
     run_bicep_format()
 
 if __name__ == "__main__":
     rich.print("Removing unecessary files")
     check_for_files()
-    rich.print("Linting files")
+    rich.print("Linting and Formatting files")
```
Comment (pamelafox) [inline hooks/post_gen_project.py:144]:
```suggestion
    rich.print("Linting and formatting files")
```

File: {{cookiecutter.__src_folder_name}}/requirements-dev.in (hunk 3 above)
Comment (pamelafox) [inline {{cookiecutter.__src_folder_name}}/requirements-dev.in:20]:
```suggestion
ruff

```

File: requirements.txt [hunk 5]
```diff
@@ -1,25 +1,21 @@
 #
-# This file is autogenerated by pip-compile with Python 3.11
+# This file is autogenerated by pip-compile with Python 3.12
 # by the following command:
 #
-#    pip-compile requirements.in
+#    pip-compile
```
Comment (pamelafox) [inline requirements.txt:5]:
```suggestion
#    pip-compile requirements.in
```

**Review by pamelafox — APPROVED** [review]

**Pull request merged**